File: 001-020/013.rb
sum = 0
# 013.data.txtを読み込む
File.open(File.expand_path('013.data.txt', __dir__), 'r') do |f|
  f.each do |line|
    # rubyでは非常に長い桁の整数をオーバーフローさせずに扱えるのでそのまま足し合わせる
    sum += line.to_i
  end
end


# 上から10桁(文字列の10文字)を取り出す
puts sum.to_s[0, 10]

File: 011-020/013.py
import os

# 問題を読み込む
sum = 0
filename = '013.data.txt'
path = os.path.join(os.path.dirname(__file__), filename)
with open(path) as f:
    for line in f.readlines():
        sum += int(line)

print(str(sum)[0:10])
